client/app/cloud/project/compute/infrastructure/virtualMachine/add/cloud-project-compute-infrastructure-virtualMachine-add.controller.js

```diff
@@ -0,0 +1,376 @@
+        this.VirtualMachineAddService = CloudProjectVirtualMachineAddService;
+        this.ovhDocUrl = ovhDocUrl;
+    }
+
+    $onInit () {
+        this.serviceName = this.$stateParams.projectId;
+        this.previousState = this.CloudNavigation.getPreviousState();
+        this.currentCurrency = this.CurrencyService.getCurrentCurrency();
+        this.loaders = {
+            adding: false
+        };
+        this.model = {
+            billingPeriod: null,
+            flavor: null,
+            imageType: null,
+            name: "",
+            networkId: "",
+            number: 0,
+            region: null,
+            sshKey: null,
+            userData: null
+        };
+        this.enums = {
+            billingPeriods: ["monthly", "hourly"],
+            flavorsTypes: [],
+            imagesTypes: []
+        };
+        this.isNameUpdated = false;
+        this.isPostScriptEnabled = false;
+        this.newSshKey = {
+            name: null,
+            publicKey: null
+        };
+        this.state = {
+            hasVRack: false
+        };
+        this.urls = {};
+    }
+
+    initProject () {
+        // Get prices in background
+        this.promisePrices = this.OvhCloudPriceHelper.getPrices(this.serviceName);
+
+        // Get quota in background
+        this.promiseQuota = this.OvhApiCloudProjectQuota.Lexi().query({ serviceName: this.serviceName }).$promise;
+
+        // Get Public Networks in background
+        this.promisePublicNetworks = this.OvhApiCloudProjectNetworkPublic.Lexi().query({ serviceName: this.serviceName }).$promise;
+
+        // Set URLs
+        this.urls.vLansApiGuide = this.ovhDocUrl.getDocUrl("g2162.public_cloud_et_vrack_-_comment_utiliser_le_vrack_et_les_reseaux_prives_avec_les_instances_public_cloud");
+        this.urls.guidesSshkey = this.ovhDocUrl.getDocUrl("g1769.creating_ssh_keys");
+    }
+
+    cancel () {
+        this.previousState.go();
+    }
+
+    confirm () {
+        this.addVirtualMachine();
+    }
+
+    /*----------------------------------
+     |  Step 1 : OS or SnapShot choice  |
+     ----------------------------------*/
+
+    initOsList () {
+        _.set(this.loaders, "step1", true);
+        return this.$q.all({
+            images: this.OvhApiCloudProjectImage.Lexi().query({ serviceName: this.serviceName }).$promise
+                .then(images => {
+                    // Image types (linux, windows, ...)
+                    this.enums.imagesTypes = this.CloudImageService.constructor.getImageTypes(images);
+                    this.images = this.VirtualMachineAddService.getAugmentedImages(images);
+                })
+                .catch(this.ServiceHelper.errorHandler("cpcivm_add_step1_images_error")),
+            snapshots: this.OvhApiCloudProjectSnapshot.Lexi().query({ serviceName: this.serviceName }).$promise
+                .then(snapshots => (this.snapshots = _.map(snapshots, snapshot => _.set(snapshot, "distribution", "linux"))))
```

Comment:
I'm not sure about this. All snapshots are linux?

Reply:
done